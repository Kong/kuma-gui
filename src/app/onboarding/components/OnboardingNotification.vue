<template>
  <!-- As well as only showing the notification on the homepage -->
  <!-- We also share the same datasource URL as the mesh listing on the homepage -->
  <DataSource
    v-if="route.name === 'home'"
    v-slot="{ data }"
    :src="`/meshes?page=1&size=10`"
  >
    <template
      v-if="data?.items.length === 1 && data.items[0].name === 'default'"
    >
      <template
        v-if="data.items[0].dataplanes.total === 0"
      >
        <div
          v-if="alertClosed === false"
          class="onboarding-check"
          data-testid="onboarding-notification"
        >
          <KAlert
            appearance="success"
            dismiss-type="icon"
            @closed="closeAlert"
          >
            <template #alertMessage>
              <div class="alert-content">
                <div>
                  <strong>Welcome to {{ t('common.product.name') }}!</strong> We've detected that you don't have any data plane proxies running yet. We've created an onboarding process to help you!
                </div>

                <div>
                  <KButton
                    appearance="primary"
                    size="small"
                    class="action-button"
                    :to="{ name: 'onboarding-welcome' }"
                  >
                    Get started
                  </KButton>
                </div>
              </div>
            </template>
          </KAlert>
        </div>
      </template>
    </template>
  </DataSource>
</template>

<script lang="ts" setup>
import { ref } from 'vue'
import { useRoute } from 'vue-router'

import { useI18n } from '@/utilities'

const { t } = useI18n()
const route = useRoute()
const alertClosed = ref(false)

function closeAlert() {
  alertClosed.value = true
}
</script>

<style lang="scss" scoped>
.onboarding-check {
  margin: 0 0 $kui-space-90 0;
}

.alert-content {
  @media screen and (min-width: 700px) {
    display: flex;
    align-items: center;

    > *:first-of-type {
      margin-right: $kui-space-60;
    }

    > *:last-of-type {
      min-width: 150px;
    }
  }

  @media screen and (max-width: 699px) {
    > *:last-of-type {
      margin-top: 10px;
    }
  }
}

.action-button.action-button {
  text-decoration: none;
}
</style>
