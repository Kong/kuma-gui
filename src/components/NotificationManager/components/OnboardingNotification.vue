<template>
  <div class="py-4">
    <KAlert appearance="success">
      <template v-slot:alertMessage>
        <div class="flex justify-between items-center">
          <div class="w-4/5">
            <strong>Welcome to {{ productName }}!</strong> We've detected that you don't have
            any data plane proxies running yet. We've created an onboarding process to
            help you!
          </div>
          <div class="w-1/5">
            <KButton
              appearance="outline-primary"
              size="small"
              class="action-button"
              :to="{ path: '/get-started' }"
            >
              Get Started
            </KButton>
          </div>
        </div>
      </template>
    </KAlert>
  </div>
</template>

<script>
import { PRODUCT_NAME } from '@/consts'

export default {
  name: 'OnboardingNotification',
  data() {
    return {
      productName: PRODUCT_NAME,
    }
  },
}
</script>

<style lang="scss" scoped>
.action-button.action-button {
  text-decoration: none;
}
</style>
