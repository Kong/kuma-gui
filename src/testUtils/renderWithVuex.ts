import { render, RenderOptions, RenderResult } from '@testing-library/vue'
import merge from 'lodash/merge'
import { StoreOptions } from 'vuex'
import Vue, { ComponentOptions } from 'vue'
import Store from '@/store'

function renderWithVuex(
  Component: typeof Vue | ComponentOptions<Vue>,
  options: RenderOptions<Vue> = { store: {}, routes: [] },
): RenderResult {
  const store: StoreOptions<any> = Store()

  // Render the component and merge the original store and the custom one
  // provided as a parameter. This way, we can alter some behaviors of the
  // initial implementation.

  return render(Component, { ...options, store: merge(store, options.store), routes: options.routes })
}

export default renderWithVuex
