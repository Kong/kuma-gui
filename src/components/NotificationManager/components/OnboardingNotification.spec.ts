import renderWithVuex from '@/testUtils/renderWithVuex'
import OnboardingNotification from './OnboardingNotification.vue'

describe('OnboardingNotification.vue', () => {
  it('renders snapshot', () => {
    const { container } = renderWithVuex(OnboardingNotification, {
      store: { modules: { config: { state: { kumaDocsVersion: '1.2.0' } } } },
      routes: [],
    })

    expect(container).toMatchSnapshot()
  })
})
