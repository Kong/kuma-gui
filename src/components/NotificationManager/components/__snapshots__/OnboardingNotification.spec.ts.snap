// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingNotification.vue renders snapshot 1`] = `
<div>
  <div
    class="onboarding-check"
  >
    <div
      class="k-alert dismissible success"
      data-v-14c8a976=""
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-v-14c8a976=""
        type="button"
      >
        <svg
          class="kong-icon kong-icon-close"
          data-v-14c8a976=""
          data-v-7d3d19ac=""
          height="12"
          role="img"
          viewBox="0 0 12 12"
          width="12"
        >
          <title
            data-v-7d3d19ac=""
          >
            close
          </title>
          <g
            data-v-7d3d19ac=""
          >
            <path
              d="M0 1l1-1h.5L6 4.49993896 10.5 0h.5l1 1v.5L7.5 6l4.5 4.5v.5l-1 1h-.5L6 7.5 1.5 12H1l-1-1v-.5L4.5 6 0 1.5z"
              data-v-7d3d19ac=""
              fill="#000"
              fill-opacity=".25"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
      <!---->
      <span
        data-v-14c8a976=""
      >
        <div
          class="alert-content"
          data-v-14c8a976=""
        >
          <div
            data-v-14c8a976=""
          >
            <strong
              data-v-14c8a976=""
            >
              Welcome to Kuma!
            </strong>
             We've detected that you don't have
          any data plane proxies running yet. We've created an onboarding process to
          help you!
        
          </div>
           
          <div
            data-v-14c8a976=""
          >
            <a
              aria-current="page"
              class="k-button action-button router-link-exact-active router-link-active small primary"
              data-v-14c8a976=""
              data-v-38fa8ecf=""
              href="#/"
              type="button"
            >
              
            Get Started
          
            </a>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
